### PR TITLE
Added FileClient.DownloadTo() methods

### LINKED ISSUE #9629: [Storage][DataLake] Add parallel download to FileClient

### PR DESCRIPTION
**Pull request by seanmcc-msft**
Resolves https://github.com/Azure/azure-sdk-for-net/issues/9629

### CODE REVIEW

File: sdk/storage/Azure.Storage.Files.DataLake/src/DataLakeFileClient.cs [hunk 1]
```diff
@@ -2105,5 +2107,249 @@ Response<Blobs.Models.BlobDownloadInfo> response
...
+        /// A <see cref="Response"/> describing the operation.
+        /// </returns>
+        /// <remarks>
+        /// A <see cref="RequestFailedException"/> will be thrown if
+        /// a failure occurs.
+        /// </remarks>
+        public virtual Response DownloadTo(
+            Stream destination,
+            DataLakeRequestConditions conditions = default,
+            //IProgress<long> progressHandler = default,
```
Comment (gapra-msft) [inline sdk/storage/Azure.Storage.Files.DataLake/src/DataLakeFileClient.cs]: Is there a plan on bringing the progress handler back?

**Review by gapra-msft — APPROVED** [review]

File: sdk/storage/Azure.Storage.Files.DataLake/src/DataLakeFileClient.cs (hunk 1 above)
Comment (amnguye) [reply sdk/storage/Azure.Storage.Files.DataLake/src/DataLakeFileClient.cs]: We had an issue for it here
https://github.com/Azure/azure-sdk-for-net/issues/8506

**Review by amnguye — APPROVED** [review]

File: sdk/storage/Azure.Storage.Files.DataLake/src/DataLakeFileClient.cs (hunk 1 above)
Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Files.DataLake/src/DataLakeFileClient.cs]: We need to implement it for downloads across all services - https://github.com/Azure/azure-sdk-for-net/issues/8506

File: sdk/storage/Azure.Storage.Files.DataLake/src/DataLakeFileClient.cs [hunk 2]
```diff
@@ -2105,5 +2107,249 @@ Response<Blobs.Models.BlobDownloadInfo> response
...
+        /// notifications that the operation should be cancelled.
+        /// </param>
+        /// <returns>
+        /// A <see cref="Response"/> describing the operation.
+        /// </returns>
+        /// <remarks>
+        /// A <see cref="RequestFailedException"/> will be thrown if
+        /// a failure occurs.
+        /// </remarks>
+        public virtual Response DownloadTo(
```
Comment (tg-msft) [inline sdk/storage/Azure.Storage.Files.DataLake/src/DataLakeFileClient.cs]: So we'll have `Read` and `DownloadTo`?  That feels confusing.  I think customers would have an easier time understanding the relationship between the two methods if they both had the same prefix like `Read` and `ReadTo`.  This should definitely get Arch Board approval as well.

File: sdk/storage/Azure.Storage.Files.DataLake/src/DataLakeFileClient.cs [hunk 3]
```diff
@@ -2105,5 +2107,249 @@ Response<Blobs.Models.BlobDownloadInfo> response
...
+            Stream destination,
+            DataLakeRequestConditions conditions = default,
+            //IProgress<long> progressHandler = default,
+            StorageTransferOptions transferOptions = default,
+            CancellationToken cancellationToken = default)
+        {
+            DiagnosticScope scope = ClientDiagnostics.CreateScope($"{nameof(Azure)}.{nameof(Storage)}.{nameof(Files)}.{nameof(DataLake)}.{nameof(DataLakeFileClient)}.{nameof(DownloadTo)}");
```
Comment (tg-msft) [inline sdk/storage/Azure.Storage.Files.DataLake/src/DataLakeFileClient.cs]: Nit - Distributed Tracing changed to drop the namespace _(an Azure wide decision that hit our repo this morning)_ so it'll be just `DiagnosticScope scope = ClientDiagnostics.CreateScope($"{nameof(DataLakeFileClient)}.{nameof(DownloadTo)}");` from here on out.

File: sdk/storage/Azure.Storage.Files.DataLake/src/DataLakeFileClient.cs [hunk 4]
```diff
@@ -2105,5 +2107,249 @@ Response<Blobs.Models.BlobDownloadInfo> response
...
+
+            try
+            {
+                scope.Start();
+
+                BlobRequestConditions blobRequestConditions = conditions.ToBlobRequestConditions();
```
Comment (tg-msft) [inline sdk/storage/Azure.Storage.Files.DataLake/src/DataLakeFileClient.cs]: This looks like a copy/paste leftover.

File: sdk/storage/Azure.Storage.Files.DataLake/src/DataLakeFileClient.cs (hunk 3 above)
Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Files.DataLake/src/DataLakeFileClient.cs]: Fixed.

File: sdk/storage/Azure.Storage.Files.DataLake/src/DataLakeFileClient.cs (hunk 4 above)
Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Files.DataLake/src/DataLakeFileClient.cs]: fixed.

**Pull request merged**